### PR TITLE
Fix - Reuse same texture for both Opaque and Blend materials And normalTexture scale / weight

### LINKED ISSUE #826: [3ds max] The normalTexture scale / weight set in my physicalmaterial does not tag along in the gltf file.
this is very related to the closed issue:
https://github.com/BabylonJS/Exporters/issues/311
But it does not solve the problems I have..

Hi, I hope I am doing something wrong.. but I cannot get the Bump map weight param from the Physical Material in max 2019 to do anything to the normalTexture scale parameter on export.
normalTextureScale-bug

From what I can see I never get that parameter written in the gltf file?
normalTextureScale-bug_p2

I can manually add the normalTexture.scale property to the gltf and then I get the behaviour that I expected.

### PR DESCRIPTION
**Pull request by pandaGaume**
Add Scale Property to GLTFTextureInfo and bind it to BabylonTexture level when != 1.0.

### CODE REVIEW

File: 3ds Max/Max2Babylon/Forms/ExporterForm.Designer.cs [hunk 1]
```diff
@@ -93,6 +93,7 @@ private void InitializeComponent()
             this.exportOptionsScrollPanel = new System.Windows.Forms.Panel();
             this.logTabPage = new System.Windows.Forms.TabPage();
             this.pictureBox1 = new System.Windows.Forms.PictureBox();
+            this.webBrowser1 = new System.Windows.Forms.WebBrowser();
```
Comment (Drigax) [inline 3ds Max/Max2Babylon/Forms/ExporterForm.Designer.cs:96]: What needed to be changed in the exporter form? I can see alot of changes to some element positioning, and a new WebBrowser component, is this needed?

File: SharedProjects/Babylon2GLTF/GLTFExporter.Material.cs [hunk 2]
```diff
@@ -387,9 +381,8 @@ private void ExportMaterial(BabylonMaterial babylonMaterial, GLTF gltf)
                     }
                 }
             }
-            else if (babylonMaterial.GetType() == typeof(BabylonPBRMetallicRoughnessMaterial))
+            else if (babylonMaterial is BabylonPBRMetallicRoughnessMaterial babylonPBRMetallicRoughnessMaterial)
```
Comment (Drigax) [inline SharedProjects/Babylon2GLTF/GLTFExporter.Material.cs:384]: Cleanup is appreciated, but its usually best to keep PRs as specific as possible, Its never an issue to make another PR to address another (seemingly) unrelated fault.

**Review by Drigax — CHANGES_REQUESTED** [review]
Are all these changes required to fix this issue? if not, then can we move these to separate PRs to address exactly what they are intended to fix?

**Comment by pandaGaume** [comment]
All this are not supposed to be commited, seems i made a mistake during the process.
1 - Nothing to change in ExporterForm, this was some experiments to understand the Script capabilities under max
2 - I will avoid to modify the code, if not related to issue

Many thanks for your feedback, i will make it better next time 🙂


Guillaume Pelletier | CEO | DotVision | + 33 6 07 42 26 30 | guillaume.pelletier@dotvision.com<mailto:guillaume.pelletier@dotvision.com>
Find us in Facebook : DotVision & DotVision Motion


________________________________
De : Nicholas Barlow <notifications@github.com>
Envoyé : jeudi 8 octobre 2020 20:41
À : BabylonJS/Exporters <Exporters@noreply.github.com>
Cc : Guillaume Pelletier <guillaume.pelletier@dotvision.com>; Author <author@noreply.github.com>
Objet : Re: [BabylonJS/Exporters] Fix #826 (#875)


@Drigax requested changes on this pull request.

Are all these changes required to fix this issue? if not, then can we move these to separate PRs to address exactly what they are intended to fix?

________________________________

In 3ds Max/Max2Babylon/Forms/ExporterForm.Designer.cs<https://github.com/BabylonJS/Exporters/pull/875#discussion_r501933542>:

> @@ -93,6 +93,7 @@ private void InitializeComponent()
             this.exportOptionsScrollPanel = new System.Windows.Forms.Panel();
             this.logTabPage = new System.Windows.Forms.TabPage();
             this.pictureBox1 = new System.Windows.Forms.PictureBox();
+            this.webBrowser1 = new System.Windows.Forms.WebBrowser();


What needed to be changed in the exporter form? I can see alot of changes to some element positioning, and a new WebBrowser component, is this needed?

________________________________

In SharedProjects/Babylon2GLTF/GLTFExporter.Material.cs<https://github.com/BabylonJS/Exporters/pull/875#discussion_r501935231>:

> @@ -387,9 +381,8 @@ private void ExportMaterial(BabylonMaterial babylonMaterial, GLTF gltf)
                     }
                 }
             }
-            else if (babylonMaterial.GetType() == typeof(BabylonPBRMetallicRoughnessMaterial))
+            else if (babylonMaterial is BabylonPBRMetallicRoughnessMaterial babylonPBRMetallicRoughnessMaterial)


Cleanup is appreciated, but its usually best to keep PRs as specific as possible, Its never an issue to make another PR to address another (seemingly) unrelated fault.

File: 3ds Max/Max2Babylon/Forms/ExporterForm.Designer.cs (hunk 1 above)
Comment (pandaGaume) [reply 3ds Max/Max2Babylon/Forms/ExporterForm.Designer.cs:96]: my mistake, nothing to change here

File: 3ds Max/Max2Babylon/Exporter/BabylonExporter.cs [hunk 3]
```diff
@@ -30,7 +30,7 @@ internal partial class BabylonExporter : ILoggingProvider
         public static string exporterVersion = "Custom.Build.Version";
         public float scaleFactorToMeters = 1.0f;
 
-        public const int MaxSceneTicksPerSecond = 4800; //https://knowledge.autodesk.com/search-result/caas/CloudHelp/cloudhelp/2016/ENU/MAXScript-Help/files/GUID-141213A1-B5A8-457B-8838-E602022C8798-htm.html
```
Comment (Drigax) [inline 3ds Max/Max2Babylon/Exporter/BabylonExporter.cs:33]: I think for the sake of explaining our magic numbers, its a good idea to always include links to autodesk documentation where appropriate.

**Comment by Drigax** [comment]
glad to hear it, can you please clean up this PR before merging?

File: 3ds Max/Max2Babylon/Exporter/BabylonExporter.cs (hunk 3 above)
Comment (pandaGaume) [reply 3ds Max/Max2Babylon/Exporter/BabylonExporter.cs:33]: Yes, however this link is broken..

**Comment by pandaGaume** [comment]
I reverse the change on my local, but it did not remove the commit from the PR.. Dit i need to use the DeleteFile command ??
[cid:d5541650-7c7a-4d88-8708-a18696b9ac68]


Guillaume Pelletier | CEO | DotVision | + 33 6 07 42 26 30 | guillaume.pelletier@dotvision.com<mailto:guillaume.pelletier@dotvision.com>
Find us in Facebook : DotVision & DotVision Motion


________________________________
De : Nicholas Barlow <notifications@github.com>
Envoyé : lundi 12 octobre 2020 21:48
À : BabylonJS/Exporters <Exporters@noreply.github.com>
Cc : Guillaume Pelletier <guillaume.pelletier@dotvision.com>; Author <author@noreply.github.com>
Objet : Re: [BabylonJS/Exporters] Fix #826 (#875)


glad to hear it, can you please clean up this PR before merging?

**Comment by Drigax** [comment]
Can you please check the diff and undo every last change that is not related to this particular fix?

**Review by Drigax — APPROVED** [review]
Fix looks good to me
